samples/bluetooth/nfc_pairing/src/main.c

```diff
@@ -0,0 +1,477 @@
+	rec_payload.flags = NFC_NDEF_LE_OOB_REC_FLAGS(BT_LE_AD_NO_BREDR);
+
+	NFC_NDEF_LE_OOB_RECORD_DESC_DEF(oob_rec, '0', &rec_payload);
+	NFC_NDEF_CH_AC_RECORD_DESC_DEF(oob_ac, NFC_AC_CPS_ACTIVE, 1, "0", 0);
+
+	return nfc_ndef_ch_msg_hs_encode(&NFC_NDEF_CH_AC_RECORD_DESC(oob_ac),
+					 &NFC_NDEF_LE_OOB_RECORD_DESC(oob_rec),
+					 1, ndef_msg_buf, len);
+}
+/** .. include_endpoint_pair_msg_rst */
+
+/**
+ * @brief Callback function for handling NFC events.
+ */
+static void nfc_callback(void *context,
+			 enum nfc_t4t_event event,
+			 const u8_t *data,
+			 size_t data_length,
+			 u32_t flags)
+{
+	ARG_UNUSED(context);
+	ARG_UNUSED(data);
+	ARG_UNUSED(flags);
+
+	switch (event) {
+	case NFC_T4T_EVENT_FIELD_ON:
+		dk_set_led_on(NFC_FIELD_LED);
+		break;
+
+	case NFC_T4T_EVENT_FIELD_OFF:
+		dk_set_led_off(NFC_FIELD_LED);
+		break;
+
+	case NFC_T4T_EVENT_NDEF_READ:
+		advertising_start();
+
+		break;
+
+	default:
+		break;
+	}
+}
+
+static void pair_msg_handler(struct k_work *work)
+{
+	int err;
+	u32_t len = sizeof(ndef_msg_buf);
+
+	err = pairing_msg_generate(&len);
+	if (err) {
+		printk("Paring message encode error\n");
+	}
+}
+
+static void advertising_continue(void)
+{
+	struct bt_le_adv_param adv_param;
+
+	bt_addr_le_t addr;
+
+	if (!k_msgq_get(&bonds_queue, &addr, K_NO_WAIT)) {
+		char addr_buf[BT_ADDR_LE_STR_LEN];
+
+		adv_param = *BT_LE_ADV_CONN_DIR(&addr);
+		adv_param.options |= BT_LE_ADV_OPT_DIR_ADDR_RPA;
+
+		int err = bt_le_adv_start(&adv_param, NULL, 0, NULL, 0);
+
+		if (err) {
+			printk("Directed advertising failed to start\n");
+			return;
+		}
+
+		bt_addr_le_to_str(&addr, addr_buf, BT_ADDR_LE_STR_LEN);
+		printk("Direct advertising to %s started\n", addr_buf);
+	} else {
+		int err;
+
+		adv_param = *BT_LE_ADV_CONN;
+		adv_param.options |= BT_LE_ADV_OPT_ONE_TIME;
+		err = bt_le_adv_start(&adv_param, ad, ARRAY_SIZE(ad),
+				      NULL, 0);
+		if (err) {
+			printk("Advertising failed to start (err %d)\n", err);
+			return;
+		}
+
+		printk("Regular advertising started\n");
+	}
+}
+
+static void adv_handler(struct k_work *work)
+{
+	advertising_continue();
+}
+
+static void auth_cancel(struct bt_conn *conn)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	printk("Pairing cancelled: %s\n", addr);
+}
+
+static void pairing_confirm(struct bt_conn *conn)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	bt_conn_auth_pairing_confirm(conn);
+
+	printk("Pairing confirmed: %s\n", addr);
+}
+
+static void lesc_obb_data_set(struct bt_conn *conn,
+			      struct bt_conn_oob_info *info)
+{
+	int err;
+
+	if (info->lesc.oob_config != BT_CONN_OOB_LOCAL_ONLY) {
+		printk("LESC OOB config not supported\n");
+		return;
+	}
+
+	/* Pass only local OOB data. */
+	err = bt_le_oob_set_sc_data(conn, &oob_local.le_sc_data, NULL);
+	if (err) {
+		printk("Error while setting OOB data: %d\n", err);
+	} else {
```

Comment:
I think this is not needed, user will know if the pairing is successful. If setting OOB returns an error, user will also be notified by the line above.

Reply:
Yes right. Removed